app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java

```diff
@@ -1028,6 +1028,14 @@ public void stop() {
         }
     }
 
+    public void refreshStream() {
+        // get current timestamp first
+        refreshCurrentPosition();
+
+        stop();
+        play(mCurrentPosition);
+    }
+
```

Comment:
For this to work with libVLC, some logic needs to be added to `clearPlaybackSessionOptions()`. `clearPlaybackSessionOptions()` is called by `stop()`.
When switching between trancoded playback and direct play, `metaVLCStreamStartPosition needs to be reset. Otherwise the timestamp will be offset by `mCurrentPosition` once direct playback starts.

https://github.com/jellyfin/jellyfin-androidtv/blob/33099abaf872c752f4136bd55c6906e2f22251e9/app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java#L1058

This was an oversight on my part since I didn't foresee a situation like this.

Changing clearPlaybackSessionOptions() to this works for me:

```

private void clearPlaybackSessionOptions() {
        mDefaultSubIndex = -1;
        mDefaultAudioIndex = -1;
        finishedInitialSeek = false;
        wasSeeking = false;
        if (mVideoManager != null)
            mVideoManager.setMetaVLCStreamStartPosition(-1);
    }

```